x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -48,46 +60,160 @@ public ModelRegistry(Client client) {
 
     public void getUnparsedModelMap(String modelId, ActionListener<ModelConfigMap> listener) {
         ActionListener<SearchResponse> searchListener = ActionListener.wrap(searchResponse -> {
+            // There should be a hit for the configurations and secrets
             if (searchResponse.getHits().getHits().length == 0) {
                 listener.onFailure(new ResourceNotFoundException("Model not found [{}]", modelId));
                 return;
             }
 
             var hits = searchResponse.getHits().getHits();
-            assert hits.length == 1;
-            listener.onResponse(new ModelConfigMap(hits[0].getSourceAsMap()));
+            listener.onResponse(createModelConfigMap(hits, modelId));
 
         }, listener::onFailure);
 
         QueryBuilder queryBuilder = documentIdQuery(modelId);
-        SearchRequest modelSearch = client.prepareSearch(InferenceIndex.INDEX_PATTERN).setQuery(queryBuilder).setSize(1).request();
+        SearchRequest modelSearch = client.prepareSearch(InferenceIndex.INDEX_PATTERN, InferenceSecretsIndex.INDEX_PATTERN)
+            .setQuery(queryBuilder)
+            .setSize(2)
+            .request();
 
         client.search(modelSearch, searchListener);
     }
 
+    private ModelConfigMap createModelConfigMap(SearchHit[] hits, String modelId) {
+        Map<String, SearchHit> mappedHits = Arrays.stream(hits).collect(Collectors.toMap(hit -> {
+            if (hit.getIndex().startsWith(InferenceIndex.INDEX_NAME)) {
```

Comment:
I wasn't sure how guarantee sorting of the documents without adding a `constant_keyword` field or something so this puts the resulting documents in a map based on the start of their `_index`